cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -58,6 +60,38 @@ def if_scvmm_refresh_provider(provider):
         provider.refresh_provider_relationships()
 
 
+def check_power_options(provider, soft_assert, vm, power_state):
+    must_be_available = {'on': [vm.POWER_OFF, vm.SUSPEND, vm.RESET],
+                         'off': [vm.POWER_ON]
+                         }
+    mustnt_be_available = {'on': [vm.POWER_ON],
+                           'off': [vm.POWER_OFF, vm.SUSPEND, vm.RESET]
+                           }
+    # VMware and RHEVM have extended power options
+    if provider.type != 'scvmm':
+        must_be_available['on'].extend([vm.GUEST_RESTART, vm.GUEST_SHUTDOWN])
+        mustnt_be_available['off'].extend([vm.GUEST_RESTART, vm.GUEST_SHUTDOWN])
+    vm.load_details()
+    toolbar.pf_select('Power')
+    for pwr_option in must_be_available[power_state]:
+        soft_assert(
+            toolbar.exists('Power', pwr_option),
```

Comment:
toolbar.exists is better as it doesn't loop UI and VM details to check each Power option